server/public/model/utils.go

```diff
@@ -304,6 +305,10 @@ func (er *AppError) Error() string {
 }
 
 func (er *AppError) Translate(T i18n.TranslateFunc) {
+	if er.SkipTranslation {
+		return
+	}
```

Comment:
This is necessary or the error message will be overridden with the ID when attempting translation which would be missing since it's coming from a plugin. The idea is that the plugin should forward the translated error if available.

Reply:
To clarify, does the i18n subsystem not just preserve the provided message if it can't find a translation?

Reply:
So, the [documented behaviour](https://github.com/mattermost/go-i18n/blob/5c415071e404a069a66863320f70b1e5a4c79f98/i18n/i18n.go#L64-L67) is to return the ID:

> // TranslateFunc returns the translation of the string identified by translationID.
//
// If there is no translation for translationID, then the translationID itself is returned.
// This makes it easy to identify missing translations in your app.

Then we are the ones eventually deciding to replace the message content with that I suppose, in the  `Translate()` function on `AppError` function above.